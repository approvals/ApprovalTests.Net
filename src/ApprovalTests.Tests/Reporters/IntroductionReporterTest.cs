using ApprovalTests.Reporters;
using NUnit.Framework;

namespace ApprovalTests.Tests.Reporters
{
    [TestFixture]
    public class IntroductionReporterTest
    {
        [Test]
        public void TestComment()
        {
            Approvals.Verify(new IntroductionReporter().GetFriendlyWelcomeMessage());
        }
    }
}