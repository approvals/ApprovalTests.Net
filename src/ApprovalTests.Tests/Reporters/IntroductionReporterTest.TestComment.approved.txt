Welcome to ApprovalTests.
====

Please add:

```
[UseReporter(typeof(DiffReporter))]
```

to your class, test method or assembly.

Why:
----

ApprovalTests uses the `[UseReporter]` attribute from your test class, method or assembly. When you do this ApprovalTest will launch the result using that reporter, for example in your diff tool.

You can find several reporters in `ApprovalTests.Reporters` namespace, or create your own by extending the `ApprovalTests.Core.IApprovalFailureReporter` interface.

Find more at: http://blog.approvaltests.com/2011/12/using-reporters-in-approval-tests.html

Best Practice:
----

Add an *assembly* level configuration. Create a file in your base directory with the name `ApprovalTestsConfig.cs`, and the contents:

```
using ApprovalTests.Reporters;

[assembly: UseReporter(typeof(DiffReporter))]
```

